execution/src/execution.cc

```diff
@@ -350,13 +350,22 @@ namespace leatherman { namespace execution {
         return execute(file, &arguments, nullptr, &environment, nullptr, stdout_callback, stderr_callback, actual_options, timeout).success;
     }
 
-    static bool process_data(bool trim, string const& data, string& buffer, string const& logger, function<bool(string&)> const& callback)
+    static bool process_data(bool trim, bool strip_cr, string const& raw_data, string& buffer, string const& logger, function<bool(string&)> const& callback)
     {
         // Do nothing if nothing was read
-        if (data.empty()) {
+        if (raw_data.empty()) {
             return true;
         }
 
+        string stripped_data;
```

Comment:
I think I have convinced myself to move this logic to `rw_from_child`, actually.